src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -27,39 +32,73 @@
     protected static final int REPORT_ID_LENGTH_FOR_DOC_NUMBER = 10;
     protected static final int NAME_LENGTH_FOR_DESCRIPTION = 14;
 
     protected String reportId;
     protected ConcurDetailLineGroupForCollectorHelper collectorHelper;
-    protected Map<String,List<ConcurStandardAccountingExtractDetailLine>> consolidatedRegularLines;
+    protected Map<String, ConcurDetailLineSubGroupForCollector> consolidatedRegularLines;
+    protected Map<String, List<ConcurStandardAccountingExtractDetailLine>> consolidatedCashAdvanceLines;
+    protected Map<String, ConcurStandardAccountingExtractDetailLine> cashAdvanceLinesByReportEntryId;
+    protected Map<String, ConcurRequestedCashAdvance> requestedCashAdvancesByCashAdvanceKey;
+    protected boolean hasMissingRequestedCashAdvance;
     protected int nextFakeObjectCode;
+    protected int nextTransactionSequenceNumber;
 
     public ConcurDetailLineGroupForCollector(String reportId, ConcurDetailLineGroupForCollectorHelper collectorHelper) {
         this.reportId = reportId;
         this.collectorHelper = collectorHelper;
         this.consolidatedRegularLines = new LinkedHashMap<>();
+        this.consolidatedCashAdvanceLines = new LinkedHashMap<>();
+        this.cashAdvanceLinesByReportEntryId = new LinkedHashMap<>();
+        this.requestedCashAdvancesByCashAdvanceKey = new LinkedHashMap<>();
+        this.hasMissingRequestedCashAdvance = false;
         this.nextFakeObjectCode = 1;
+        this.nextTransactionSequenceNumber = 1;
     }
 
     public String getReportId() {
         return reportId;
     }
 
     public void addDetailLine(ConcurStandardAccountingExtractDetailLine detailLine) {
-        String accountingFieldsKey = buildAccountingFieldsKey(detailLine);
-        consolidatedRegularLines.computeIfAbsent(accountingFieldsKey, (key) -> new ArrayList<>())
-                .add(detailLine);
+        if (isCashAdvanceLine(detailLine)) {
+            ConcurRequestedCashAdvance requestedCashAdvance = requestedCashAdvancesByCashAdvanceKey.computeIfAbsent(
+                    detailLine.getCashAdvanceKey(), this::getExistingRequestedCashAdvanceByCashAdvanceKey);
+            String accountingFieldsKey = buildAccountingFieldsKeyForCashAdvance(detailLine, requestedCashAdvance);
+            consolidatedCashAdvanceLines.computeIfAbsent(accountingFieldsKey, (key) -> new ArrayList<>())
+                    .add(detailLine);
+            cashAdvanceLinesByReportEntryId.put(detailLine.getReportEntryId(), detailLine);
+        } else {
+            String accountingFieldsKey = buildAccountingFieldsKey(detailLine);
+            consolidatedRegularLines.computeIfAbsent(accountingFieldsKey, ConcurDetailLineSubGroupForCollector::new)
+                    .addDetailLine(detailLine);
+        }
+    }
+
+    protected boolean isCashAdvanceLine(ConcurStandardAccountingExtractDetailLine detailLine) {
+        return StringUtils.isNotBlank(detailLine.getCashAdvanceKey());

```

Comment:
Good catch, I had started implementing the use of your new service method but I didn't completely change things over in this class. I'll make that change.